docs/guides/typescript.md

```diff
@@ -233,3 +203,258 @@ type State =
 ```
 
 :::
+
+## Troubleshooting
+
+There are some known limitations with XState and TypeScript. We love TypeScript, and we're _constantly_ pressing ahead to make it a better experience in XState.
+
+Here are some known issues, all of which can be worked around:
+
+### Events in machine options
+
+When you use `createMachine`, you can pass in implementations to named actions/services/guards in your config. For instance:
+
+```ts
+interface Context {}
+
+type Event =
+  | { type: 'EVENT_WITH_FLAG'; flag: boolean }
+  | {
+      type: 'EVENT_WITHOUT_FLAG';
+    };
+
+createMachine<Context, Event>(
+  {
+    on: {
+      EVENT_WITH_FLAG: {
+        actions: 'consoleLogData'
+      }
+    }
+  },
+  {
+    actions: {
+      flag: (context, event) => {
```

Comment:
Oh cool, I can comment here. Haven't done this before. So I'll be a little verbose as I enjoy the experience. 😂

Should this action be named `consoleLogData`, or should you be calling the action named `flag` up on line 230?

Reply:
See also below.

Reply:
It should! Fixed this in a later PR, though not the one below! Good eyes, sir